### DISCUSSION TITLE
Adding vendors

### DISCUSSION THREAD

**Discussion by Merlin246** [supermerill/SuperSlicer#1854] [Q&A]
Hello,

Newby to superslicer so forgive any ignorance.

I have an E3D Toolchanger I am trying to add as a printer profile, ideally I'd like to contribute so E3D is a vendor allowing others a more streamlined installation process. I have compiled the thumbnail, bed stl & svg files. I added them into a new E3D folder under resources > profiles > E3D as well as the .ini config file from [E3D's github](https://github.com/e3donline/Slicer-Profiles/tree/master/SuperSlicer) to the profiles folder. When trying to add a printer the program crashes with the error: 

Internal error: Vendor Config Bundle 'E3D' is not valid: Missing secion or key: 'vendor'.

I'm guessing vendors need to be added to a master list in a file somewhere, but I do not actually know. Is there a way for individuals to add vendors easily?

**Comment by neophyl** [comment]
AFAIK vendors/profiles are located at https://github.com/slic3r/slic3r-profiles  which is part of the base Slic3r project.

**Comment by supermerill** [comment]
The simple way to begin to write its own vendor profile is to duplicate a simple one, like the luzbot.
The vendor ini has some mor field to add. They are for most of them pretty self-explanatory.
there is the header, then the printer list
then you have to declare your print, printer, filament 'pattern', and then the real one that can inherit one or multiple patterns, and add some modifications.
You have to do the same for each variant (often used for nozzle size, but you can use a variant for whatever you want)

You can look at the creality or voron to have a good example of usage of inheritance and patterns.

when starting the slicer in console mode, it will give you error message when something is wrong with your vendor. Began simple and test often to reduce bug hunting time.

**Reply by mlee12382** [reply]
If we already have vendor profile set up can we submit it for addition to SS or would we need to submit it to PS using their system and then wait for it to trickle through to SS through the normal update process?

**Reply by supermerill** [reply]
submit it to https://github.com/slic3r/slic3r-profiles
You can also submit it to prusaslicer.
Be careful, PS profiles and SuSi profile are incompatibles. SuSI contains fucntions to convert them.
If you want to convert a vendor profile from PS, try applying https://github.com/slic3r/slic3r-
profiles/blob/748fbdfd2ac077e4e415868e7bc963740b92aa8e/translate_from_prusa.py
And there should be also some manual adjustments you may want to add, to activate some SuperSlicer functions.

**Reply by mlee12382** [reply]
Awesome, thanks  :) I submitted to PS last night, I will look into these also :)

**Comment by DzzD** [comment]
I would also like to add a new Vendor "Geeetech" with at least  one printer A20M+Purgebucket option and a new hybrid printer for another vendor Creality (CR10SPro + BondTechDDX + CHT Nozzle), but I still not understand all about notes, dependencies, vendor ...
